spec/models/target_spec.rb

```diff
@@ -32,74 +35,130 @@
+  end
 
-    it 'returns 10 if there are 30 program minutes of 300' do
-      conference = create(:conference)
-      target = create(:target, target_count: 300, unit: Target.units[:program_minutes])
-      event = create(:event)
+  describe '#get_campaign' do
+    context 'submissions' do
+      before do
+        submission_target.campaign = create(:campaign, name: 'Submission Campaign', conference: submission_target.conference)
+        submission_target.created_at = Time.utc(2014, 5, 10)
+        submission_target.due_date = Date.today + 4.days
+        allow(submission_target.campaign).to receive(:submissions_count) { 20 }
+      end
+
+      it 'returns a hash with values of the corresponding campaign submissions' do
+        result = {
+          'target_name' => "10 Submissions by #{Date.today + 4.days}",
+          'campaign_name' => 'Submission Campaign',
+          'value' => 20,
+          'unit' => 'Submission',
+          'created_at' => Time.utc(2014, 5, 10).in_time_zone,
+          'progress' => '200',
+          'days_left' => 4
+        }
+
+        expect(submission_target.get_campaign).to eq result
+      end
+    end
 
-      conference.targets = [target]
-      conference.program.events = [event]
+    context 'registrations' do
+      before do
+        registration_target.campaign = create(:campaign, name: 'Registration Campaign', conference: registration_target.conference)
+        registration_target.created_at = Time.utc(2014, 5, 10)
+        registration_target.due_date = Date.today + 4.days
+        allow(registration_target.campaign).to receive(:registrations_count) { 20 }
+      end
+
+      it 'returns a hash with values of the corresponding campaign registrations' do
+        result = {
+          'target_name' => "10 Registrations by #{Date.today + 4.days}",
+          'campaign_name' => 'Registration Campaign',
+          'value' => 20,
+          'unit' => 'Registration',
+          'created_at' => Time.utc(2014, 5, 10).in_time_zone,
+          'progress' => '200',
+          'days_left' => 4
+        }
+
+        expect(registration_target.get_campaign).to eq result
+      end
+    end
 
-      expect(target.get_progress).to eq('10')
+    context 'program_minutes' do
+      before do
+        program_minutes_target.campaign = create(:campaign, name: 'Program Campaign', conference: program_minutes_target.conference)
+        program_minutes_target.created_at = Time.utc(2014, 5, 10)
+        program_minutes_target.due_date = Date.today + 4.days
+        allow(program_minutes_target.conference).to receive(:current_program_minutes) { 20 }
+      end
+
+      it 'returns a hash with values of the corresponding campaign program minutes' do
+        result = {
+          'target_name' => "300 Program minutes by #{Date.today + 4.days}",
+          'campaign_name' => 'Program Campaign',
+          'value' => 20,
+          'unit' => 'Program minute',
+          'created_at' => Time.utc(2014, 5, 10).in_time_zone,
```

Comment:
I actually can't remember what this does. @ChrisBr might as this is his code :-)